streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorTopology.java

```diff
@@ -59,4 +78,78 @@ public SourceNode source(String topic) {
         return stateStoreSuppliers;
     }
 
+
+    /**
+     * Produces a string representation contain useful information about a ProcessorTopology.
```

Comment:
`.. about this topology by performing a breadth-first traversal of the DAG from source nodes. This is useful in debugging scenarios`.